### PR TITLE
Update Makefile for terminfo and create xterm-256color terminfo

### PR DESCRIPTION
**Pull request by tmquickfds**

### CODE REVIEW

**Comment by geneb** [comment]
@WolWise1 Can you validate these changes for me?  (there's merge conflicts with this one)

**Comment by tmquickfds** [comment]
Hi,

Yes, there is a conflict with #68. I have written another solution to remove terminfo directory before copy a new one, and make a new term definition. But I think it is best to user "-f" option when is needed.

For pcode, the problem is to recompile it you should use ScarletDME BASIC command. But to use ScarletDME you should have a pcode.

Regards.
Thierry

**Comment by WolWise1** [comment]
On 08/12/2023 06:54, tmquickfds wrote:
> Hi,
> 
> Yes, there is a conflict with #68 
> <https://github.com/geneb/ScarletDME/pull/68>. I have written another 
> solution to remove terminfo directory before copy a new one, and make a 
> new term definition. But I think it is best to user "-f" option when is 
> needed.

Yup. I guess it failed when trying to remove terminfo/ - as it contains 
subdirectories.

But if you're taking that approach, I need to use that technique for 
NEWVOC, MESSAGES, maybe a couple more. terminfo should be handled 
similarly, ie delete and rebuild /git/.../qmsys/terminfo, and then 
include terminfo with the other two directories to copy live.
> 
> For pcode, the problem is to recompile it you should use ScarletDME 
> BASIC command. But to use ScarletDME you should have a pcode.
> 
In other words, catch-22. I thought so :-(

There will be a lot of pressure on us to remove that binary blob, and if 
I'm going to try and get it to build on Windows, I might have to ...

Cheers,
Wol

**Comment by geneb** [comment]
On Thu, 7 Dec 2023, WolWise1 wrote:

> On 07/12/2023 20:16, Gene Buckle wrote:
>> @WolWise1 <https://github.com/WolWise1> Can you validate these changes
>> for me? (there's merge conflicts with this one)
>>
> The changes look good. The terminfo change has basically removed
> terminfo from the list of targets required by the qm target. I think
> that's probably down to me removing the terminfo target, and not
> tripping over it when testing.
>
Ok.

> With pcode, I'm not happy about it being left, but I'm guessing that's
> because we don't have the source to rebuild it?
>
We actually DO have the source to rebuild it, except for one or two 
modules that got pulled from the GPL release for whatever reason. (They'd 
make good targets for a decompiler I suspect.)  I'm pretty sure I wrote up 
the pcode stuff in the notes.txt file...

> And likewise defining RM as "rm -f" - I'd be happier adding the -f where
> rm is actually used and it's needed, but I don't know how makefiles are
> typically done. Maybe that's normal.
>
That makes me nervous - it's a recipe for disaster I think.

> Here's hoping I get a chance to hack over Christmas ...
>

Hehe.  I've got tiny, tiny, fragments of spare time these days. :(

g.

-- 
Proud owner of F-15C 80-0007
http://www.f15sim.com - The only one of its kind.
http://www.diy-cockpits.org/coll - Go Collimated or Go Home.
Some people collect things for a hobby.  Geeks collect hobbies.

ScarletDME - The red hot Data Management Environment
A Multi-Value database for the masses, not the classes.
http://scarlet.deltasoft.com - Get it _today_!
